### ISSUE TITLE
Need information before I commit to this plugin

### ISSUE THREAD

**Issue by vaughn-taylor** [nystudio107/craft3-transcoder#19]
@khalwat, before I move forward with this plugin I just need to check off a few boxes regarding functionality. I read through the documentation, but I may not completely understand the plugin (I apologize in advance for stupid questions):

1. Can I use this plugin to convert an uploaded mp4 to my other HTML5 video fallbacks -- **webm** and **ogg**? If yes, can the other formats be converted all at once with a single mp4 upload?
2. My videos are currently uploaded and stored in an Amazon S3 bucket. If I upload an mp4, can the conversions be stored in the same S3 bucket?

The bottom line is that I need a convenient way to get those HTML5 video fallbacks. For my clients, I'd like to avoid hosting with YT or Vimeo and instead host videos on S3. I want to make it easy for clients to upload a single video knowing that it will function across various browsers.

**Comment by khalwat** [comment]
1. Yes, you'd just create multiple calls to convert to the formats you want
2. Not currently; it can read _from_ an S3 or other source, but it currently writes the videos locally, and does not have support for automatic uploading to an S3 bucket

**Comment by internetztube** [comment]
Support for uploading to a Storage Bucket would be awesome.

**Comment by khalwat** [comment]
Feel free to file a FR!

**Comment by internetztube** [comment]
@khalwat Would it be ok if I would rewrite the entire plugin so that the assets are only stored in volumes?

**Comment by khalwat** [comment]
You're welcome to do that if you like. You're thinking about abstracting it away to use Craft's `volume`s @internetztube ?

**Comment by internetztube** [comment]
Yea. Check out my Pull Request. @khalwat
